src/Orleans.Core.Abstractions/Runtime/GrainReference.cs

```diff
@@ -330,7 +346,15 @@ protected void InvokeOneWayMethod(int methodId, object[] arguments, InvokeMethod
         /// </summary>
         protected Task<T> InvokeMethodAsync<T>(int methodId, object[] arguments, InvokeMethodOptions options = InvokeMethodOptions.None, SiloAddress silo = null)
         {
-            return this.Runtime.InvokeMethodAsync<T>(this, methodId, arguments, options | invokeMethodOptions, silo);
+            try
+            {
+                OrleansGrainReferenceEvent.Log.InvokeMethodAsyncStart();
+                return this.Runtime.InvokeMethodAsync<T>(this, methodId, arguments, options | invokeMethodOptions, silo);
+            }
+            finally
+            {
+                OrleansGrainReferenceEvent.Log.InvokeMethodAsyncStop();
+            }
```

Comment:
The *Stop event needs to be emitted once the task completes, otherwise I think it breaks activity support

Reply:
I see , let me do that